### ISSUE TITLE
Stuck on loading users and loading boards

### ISSUE THREAD

**Issue by kyam808** [kiswa/TaskBoard#289]
I'm able to login to taskboard but when I try to go into the settings it gets stuck on Loading Users and Loading Boards. 
![capture](https://cloud.githubusercontent.com/assets/19560072/15521270/8c6c4b8e-21a6-11e6-943f-b4eecff65bfe.JPG)

Opening up developer tools and looking at the console returns these errors:
TypeError: Unable to get property 'options' of undefined or null reference
   at loadOptionsData (http://localhost/taskboard-master/js/controllers/settings.js:31:9)
   at Anonymous function (http://localhost/taskboard-master/js/controllers/settings.js:24:13)
   at Anonymous function (http://localhost/taskboard-master/lib/angular.js:9408:11)
   at processQueue (http://localhost/taskboard-master/lib/angular.js:13248:11)
   at Anonymous function (http://localhost/taskboard-master/lib/angular.js:13264:27)
   at Scope.prototype.$eval (http://localhost/taskboard-master/lib/angular.js:14466:9)
   at Scope.prototype.$digest (http://localhost/taskboard-master/lib/angular.js:14282:15)
   at Scope.prototype.$apply (http://localhost/taskboard-master/lib/angular.js:14571:13)
   at done (http://localhost/taskboard-master/lib/angular.js:9698:36)
   at completeRequest (http://localhost/taskboard-master/lib/angular.js:9888:7)

I'm pretty noob when it comes to any web/coding, so if possible ELI5 =) Hopefully this is an easy fix. Let me know if I need to provide more information.

I'm running Apache 2.4 with PHP 5.6.21 on Windows 7.

Thanks in advance!

**Comment by kyam808** [comment]
I read some of the issues other people had with loading users and boards. Apparently when I try to access http://localhost/taskboard-master/api/ it returns this:

Warning: require_once(C:\Apache24\htdocs\TaskBoard-master\api): failed to open stream: Permission denied in C:\Apache24\htdocs\TaskBoard-master\api\api.php on line 3

Fatal error: require_once(): Failed opening required 'C:\Apache24\htdocs\TaskBoard-master\api' (include_path='.;C:\php\pear') in C:\Apache24\htdocs\TaskBoard-master\api\api.php on line 3

I'm guessing I didn't set it up properly? I do have AllowOverride all set in <Directory "c:/Apache24/htdocs">

Any guidance in the right direction would be greatly appreciated.

**Comment by kiswa** [comment]
It sounds like the JS is breaking when trying to load the default user options that should have been created for the default admin user.

Is this how it's behaving immediately after a fresh install? If so, give the `dev` branch a try and see if it works for you.

**Comment by kyam808** [comment]
I installed the dev branch and installed the dependencies but the same issue persists where it gets stuck in loading. When I view http://localhost/taskboard-dev/api/ it displays a new error message:

Warning: require_once(C:\Apache24\htdocs\TaskBoard-dev\api/../vendor/autoload.php): failed to open stream: No such file or directory in C:\Apache24\htdocs\TaskBoard-dev\api\api.php on line 3

Fatal error: require_once(): Failed opening required 'C:\Apache24\htdocs\TaskBoard-dev\api/../vendor/autoload.php' (include_path='.;C:\php\pear') in C:\Apache24\htdocs\TaskBoard-dev\api\api.php on line 3

Line 3 in api.php = require_once(__DIR__.'/../vendor/autoload.php');
Do I need to put the full path? I'm not too familiar with coding.

**Comment by kiswa** [comment]
That looks like the vendor files were not installed. See [step 2.](https://github.com/kiswa/TaskBoard#install) of the Installation in the README.

> Open `TaskBoard/` in a terminal and run `./build/composer.phar install`

**Comment by kyam808** [comment]
Oh that was weird. in the prod one Apache was okay where the vendor files got defaulted to. Anyway I moved the vendor install files to the taskboard-dev folder but now I can't even login to TaskBoard. I login as admin but it won't let me in. Pulling up console returns the following errors:

TypeError: Object expected
   at Anonymous function (http://localhost/taskboard-dev/lib/angular.js:9415:11)
   at processQueue (http://localhost/taskboard-dev/lib/angular.js:13248:11)
   at Anonymous function (http://localhost/taskboard-dev/lib/angular.js:13264:27)
   at Scope.prototype.$eval (http://localhost/taskboard-dev/lib/angular.js:14466:9)
   at Scope.prototype.$digest (http://localhost/taskboard-dev/lib/angular.js:14282:15)
   at Scope.prototype.$apply (http://localhost/taskboard-dev/lib/angular.js:14571:13)
   at done (http://localhost/taskboard-dev/lib/angular.js:9698:36)
   at completeRequest (http://localhost/taskboard-dev/lib/angular.js:9888:7)
   at requestLoaded (http://localhost/taskboard-dev/lib/angular.js:9829:9)

**Comment by kyam808** [comment]
Not sure if this helps too but when I check the network in developer tools it looks like it's doing a bunch of get requests to directories that don't exist.

![capture](https://cloud.githubusercontent.com/assets/19560072/15621601/b6fce5f8-23fd-11e6-99cb-f38617a57135.JPG)

![capture2](https://cloud.githubusercontent.com/assets/19560072/15621614/c8df65f2-23fd-11e6-9dad-6b220cda0282.JPG)

**Comment by kiswa** [comment]
Yeah, those are not actual directories. The `.htaccess` rewrites those requests to the `api.php` file so it can route appropriately.

The object expected error is a front-end issue, but I have no idea what's causing it. Try this: hit <kbd>F12</kbd> and go to the Resources tab (that's what it's called in Chrome anyway) then find Local Storage and delete anything it has for your TaskBoard domain. Then try logging in again and let me know what happens.

**Comment by kyam808** [comment]
Sorry for the delayed response. Is there a way to clear the Local Storage in IE? Developer Tools are locked down in Chrome currently.

**Comment by kiswa** [comment]
I'm sure there is, but I don't currently have access to IE (I'm on Linux). I would imagine it's similar to the steps for Chrome.

**Comment by kiswa** [comment]
Closing due to inactivity. Let me know if it needs to be re-opened.
